src/test/java/com/amazonaws/encryptionsdk/kmssdkv2/AwsKmsMrkAwareMasterKeyProviderTest.java

```diff
@@ -256,7 +255,7 @@ public void always_need_a_region() {
     // # obtained initialization MUST fail.
     public void discovery_region_can_not_be_null() {
       assertThrows(
-          IllegalArgumentException.class,
+          AwsCryptoException.class,
```

Comment:
This test would never throw an exception, even though we set `discoveryMrkRegion` set to `null` in builder. It will try to fetch Default region from Credentials Provider chain with the new changes.

Reply:
Oh, your right, I went too far.
We should revert this:
https://github.com/aws/aws-encryption-sdk-java/blob/69cfee6cfc9403335925a35fd35717aecce94c7d/src/main/java/com/amazonaws/encryptionsdk/kmssdkv2/AwsKmsMrkAwareMasterKeyProvider.java#L170-L188